### PR TITLE
Profiling vignette

### PR DESCRIPTION
**Pull request by jgabry**
Note: this branch branches off of the profiling branch (so it looks like there are more files changed than there really are)

@rok-cesnovar Here's a template for the profiling vignette. To generate the html run: 

```r
withr::with_envvar(c("NOT_CRAN" = "true"), {
  pkgdown::build_articles_index() 
  pkgdown::build_article("profiling")
})
```

### CODE REVIEW

File: _pkgdown.yml [hunk 1]
```diff
@@ -65,10 +65,11 @@ articles:
   - title: "More details"
     desc: >
       More information about compilation, passing in data,
-      how CmdStan ouput is written to CSV and read back into R, and
-      using CmdStanR in R Markdown documents.
+      how CmdStan ouput is written to CSV and read back into R,
+      profiling Stan programs, and using CmdStanR in R Markdown documents.
     contents:
       - cmdstanr-internals
+      - profiling
```
Comment (jgabry) [inline _pkgdown.yml:72]: When creating a new vignette you need to add the file name (without the .Rmd) to the articles section of _pkgdown.yml

File: vignettes/profiling.Rmd [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+params:
+  EVAL: !r identical(Sys.getenv("NOT_CRAN"), "true")
+vignette: >
+  %\VignetteIndexEntry{Profiling Stan programs with CmdStanR}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r child="children/settings-knitr.Rmd"}
+```
```
Comment (jgabry) [inline vignettes/profiling.Rmd:17]: brings in the knitr settings we use in all the vignettes

File: vignettes/profiling.Rmd [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  rmarkdown::html_vignette:
+    toc: true
+    toc_depth: 4
+params:
+  EVAL: !r identical(Sys.getenv("NOT_CRAN"), "true")
+vignette: >
+  %\VignetteIndexEntry{Profiling Stan programs with CmdStanR}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
```
Comment (jgabry) [inline vignettes/profiling.Rmd:14]: I just copied all this from the other vignettes (but I put your name first in the authors)

File: vignettes/profiling.Rmd [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Accessing profiling information after running Stan
+
+### Load profile data into R
+
+The `$profiles()` method ...
+
+### Saving profile files
+
+The `$save_profile_files()` method ...
```
Comment (jgabry) [inline vignettes/profiling.Rmd:36]: Feel free to change these section titles and order. Just wanted to show that we've been using ## for the main sections (# is used automatically for the title) and ### for subsections (#### for subsubsections if necessary).

**Comment by jgabry** [comment]
Note: I made this as a PR so I could add review comments

**Comment by rok-cesnovar** [comment]
Thank you so much!

**Comment by jgabry** [comment]
Oh and after you run 

```r
withr::with_envvar(c("NOT_CRAN" = "true"), {
  pkgdown::build_articles_index() 
  pkgdown::build_article("profiling")
})
```

you can run 

```r
pkgdown::preview_site()
```

to see what it looks like

**Comment by bbbales2** [comment]
@jgabry two questions:

1. I added the Stan models as files (instead of being inline in the vignette). Where do those go? I left them in the vignettes folder.
2. I wanna add a workflow citation to the beginning and am too lazy to look this up. Do you know how off hand? Will I just make a .bib somewhere? If not, I will dig more. Just being lazy.

**Comment by jgabry** [comment]
> 1. I added the Stan models as files (instead of being inline in the vignette). Where do those go? I left them in the vignettes folder.

That's fine, although I would slightly prefer creating a separate folder (inside the vignettes folder) for those. 

> 2. I wanna add a workflow citation to the beginning and am too lazy to look this up. Do you know how off hand? Will I just make a .bib somewhere? If not, I will dig more. Just being lazy.

Yeah there's an example in the rstanarm vignettes: 

https://github.com/stan-dev/rstanarm/tree/master/vignettes

Check out `mrp.Rmd`, which references the `.bib` file in the separate `mrp-files` folder: 

https://github.com/stan-dev/rstanarm/blob/dee0a2d45bf42b2df791072041151b753edd6af9/vignettes/mrp.Rmd#L8

and then includes the citations like this: 

https://github.com/stan-dev/rstanarm/blob/dee0a2d45bf42b2df791072041151b753edd6af9/vignettes/mrp.Rmd#L30-L38

**Comment by rok-cesnovar** [comment]
Tests pass, should be ready for review.

**Comment by jgabry** [comment]
Cool, taking a look now

**Comment by jgabry** [comment]
> Tests pass, should be ready for review.

Technically I opened this PR so it won't let me add a review. But I pushed a few minor edits and I think this is ready to go. The only thing is that I think it will fail on testcoverage because that runs the vignettes but doesn't have the release candidate.

**Review by rok-cesnovar — APPROVED** [review]
Approving for @jgabry

**Comment by rok-cesnovar** [comment]
Thanks. Should pass now.

**Pull request merged**